### ISSUE TITLE
Error handling: finish headline cases

### ISSUE THREAD

**Issue by josephjclark** [OpenFn/kit#455]
There are a bunch more error cases I need to crap and test for:

* Syntax Errors (right now I think we get lost in the compiler, so these break the whole platform)
* Timeouts
* OOM (can I test this?)
* Security/eval
* Exceptions
* Adaptor errors
* Workflow valdation errors
* Various Import errors in the linker

I've also got to check that kill reasons are properly tracked (right now it doesn't look like it)

Basically every error in the spreadsheet must be tracked

**Comment by josephjclark** [comment]
Closing this issue down.

Two of the big ones, OOM and Timeout, are being well handled and have had tests added over the last couple of weeks.
